server.js

```diff
@@ -449,12 +452,16 @@ function Protocols() {
                     }
                 }
 
+                if (!this.objectData || !this.objectData.data) {
+                    console.warn('protocol: Unable to find objectData', message);
+                }
+
                 return {
                     object: msgContent.object,
                     frame: msgContent.frame,
                     node: msgContent.node,
                     logic: msgContent.logic,
-                    data: this.objectData.data
+                    data: this.objectData.data || msgContent.data,
```

Comment:
Since this doesn't return early if !this.objectData, should it be something like `(this.objectData && this.objectData.data) || msgContent.data` (similar to what you did in messageToSend)?

Reply:
Good point! Updated accordingly